Formula/hyperscan.rb

```diff
@@ -15,11 +15,21 @@ class Hyperscan < Formula
   depends_on "cmake" => :build
   depends_on "pkg-config" => :build
   depends_on "ragel" => :build
+  # Only supports x86 instructions and will fail to build on ARM.
+  # See https://github.com/intel/hyperscan/issues/197
+  depends_on arch: :x86_64
   depends_on "pcre"
 
   def install
+    cmake_args = std_cmake_args + ["-DBUILD_STATIC_AND_SHARED=ON"]
+    on_linux do
+      # Linux CI cannot guarantee AVX2 support needed to build fat runtime.
+      # As this will limit bottle to single microarch, we force older SSSE3 for compatibility.
+      cmake_args << "-DFAT_RUNTIME=OFF" << "-DHAVE_AVX2=OFF"
```

Comment:
@Homebrew/linux, wanted to check if using lowest common microarch is preferred, or if there is a CI runner to allow building multi-microarch support (so AVX2 can be used when available)

Reply:
It's fine to disable all AVX things for the moment. We could have a self-hosted runner with AVX, but this also might need 2 different bottles, one for each microarch. Some libraries detect and use the micro arch at runtime; but this is not always the case, so we would break the bottle for users who do not have AVX.

Reply:
I'm wondering if this formula has other ISAs (i.e. SSE4) that we don't technically require (minimum ISA for bottles is SSSE3) which would be broken as well for users on older hardware?

Reply:
> I'm wondering if this formula has other ISAs (i.e. SSE4) that we don't technically require

`hyperscan` build scripts only check `SSSE3` and `AVX2`, but the code may optimize for `SSE4`. e.g. https://github.com/intel/hyperscan/blob/v5.4.0/src/util/simd_utils.h#L117-L123
```h
#if defined(HAVE_SSE41)
    a = _mm_cmpeq_epi64(a, b);
    return ~(_mm_movemask_ps(_mm_castsi128_ps(a))) & 0x5;
#else
    u32 d = diffrich128(a, b);
    return (d | (d >> 1)) & 0x5;
#endif
```

I may need to check if we need to disable some of these.

Reply:
It's too bad this formula doesn't do what OpenBLAS does, for example, where it builds up to the newest ISA and then does run time checks to see which ISAs are actually supported.  That means you can build on whatever system you want, and all users will get the most optimized ISA they run.

Reply:
https://github.com/intel/hyperscan/blob/master/doc/dev-reference/getting_started.rst#target-architecture

What `-march` do we use for Linux builds?

Reply:
We use `core2` in CI.

Reply:
Then it is should be fine. Probably can remove the `-DHAVE_AVX2=OFF` as I think the test code to check this will be compiled with `-march=core2` and probably fail.

Reply:
Great, that appears to have worked on Linux.